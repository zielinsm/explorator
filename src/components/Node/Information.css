.Information__Header {
    padding-top: 0 !important;
    margin-bottom: 5px !important;
}
