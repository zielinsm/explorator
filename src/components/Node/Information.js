import React from "react";
import { Header, Divider, Label, Tab as Switch } from "semantic-ui-react";
import filesize from "filesize";
import "./Information.css";

import Tab from "./Tab";

const Information = props => {
  const { results } = props;

  const {
    name,
    id,
    or_addr,
    dir_addr,
    details,
    country,
    as,
    weight,
    host,
    bandwidth,
    policy,
    family_alleged,
    family_effective,
    info,
    tor
  } = results;

  const basicTab = [
    { label: "OR Addresses", value: or_addr },
    { label: "Directory Address", value: dir_addr },
    { label: "Hostnames", value: host },
    { label: "Country", value: country },
    { label: "AS Name", value: as.name },
    { label: "Consensus weight", value: weight },
    { label: "Platform", value: tor },
    { label: "Contact", value: info }
  ];

  const bandwidthTab = [
    { label: "Bandwidth Rate", value: `${filesize(bandwidth.rate)}/s` },
    { label: "Bandwidth Burst", value: `${filesize(bandwidth.burst)}/s` },
    {
      label: "Observed Bandwidth",
      value: `${filesize(bandwidth.observed)}/s`
    },
    {
      label: "Advertised Bandwidth",
      value: `${filesize(bandwidth.advertised)}/s`
    }
  ];

  const policiesTab = [{ label: "Exit policy", value: policy }];

  const familyTab = [
    { label: "Alleged family", value: family_alleged },
    { label: "Effective family", value: family_effective }
  ];

  const panes = [
    {
      menuItem: "Basic",
      render: () => <Tab data={basicTab} />
    },
    {
      menuItem: "Bandwidth",
      render: () => <Tab data={bandwidthTab} />
    },
    {
      menuItem: "Policies",
      render: () => <Tab data={policiesTab} />
    },
    {
      menuItem: "Family",
      render: () => <Tab data={familyTab} />
    }
  ];

  return (
    <div>
      <Header
        as="h3"
        content={name}
        subheader={id}
        className="Information__Header"
      />
      {details.map((element, index) => (
        <Label size="tiny" key={index}>
          {element}
        </Label>
      ))}
      <Divider />
      <Switch menu={{ secondary: true }} panes={panes} />
    </div>
  );
};

export default Information;
