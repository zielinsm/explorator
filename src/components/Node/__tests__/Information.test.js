import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Information from "../Information";

const mockResults = {
  name: "Nickname",
  id: "FINGERPRINT",
  or_addr: ["000.000.000.000:9000"],
  dir_addr: "000.000.000.000:9001",
  up: true,
  details: ["Mock", "Running"],
  country: "Mock",
  as: {
    number: "AS194",
    name: "Mock Networks"
  },
  weight: 10000,
  host: "mock-relay.net",
  bandwidth: {
    rate: 10000,
    burst: 10000,
    observed: 10000,
    advertised: 10000
  },
  policy: ["reject *:*"],
  policy_summary_v4: { reject: ["1-65535"] },
  policy_summary_v6: { reject: ["1-65535"] },
  family_alleged: ["ALLEGED"],
  family_effective: ["EFFECTIVE"],
  info: "Contact",
  tor: "Tor 0.3.2.10 on Mock"
};

describe("Information component:", () => {
  const component = <Information results={mockResults} />;

  it("renders without crashing", () => {
    const wrapper = shallow(component);
    expect(wrapper.find(".Information__Header")).toHaveLength(1);
  });

  it("snapshot matches", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
