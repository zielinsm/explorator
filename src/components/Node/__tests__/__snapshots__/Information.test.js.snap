// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information component: snapshot matches 1`] = `
<div>
  <h3
    className="ui header Information__Header"
  >
    Nickname
    <div
      className="sub header"
    >
      FINGERPRINT
    </div>
  </h3>
  <div
    className="ui tiny label"
    onClick={[Function]}
  >
    Mock
  </div>
  <div
    className="ui tiny label"
    onClick={[Function]}
  >
    Running
  </div>
  <div
    className="ui divider"
  />
  <div>
    <div
      className="ui secondary menu"
    >
      <a
        className="active item"
        onClick={[Function]}
      >
        Basic
      </a>
      <a
        className="item"
        onClick={[Function]}
      >
        Bandwidth
      </a>
      <a
        className="item"
        onClick={[Function]}
      >
        Policies
      </a>
      <a
        className="item"
        onClick={[Function]}
      >
        Family
      </a>
    </div>
    <div
      className="row Tab__row"
    >
      <div
        className="ui sub header"
      >
        OR Addresses
      </div>
      <span
        className="Tab__field"
      >
        000.000.000.000:9000 
        
      </span>
    </div>
    <div
      className="row Tab__row"
    >
      <div
        className="ui sub header"
      >
        Directory Address
      </div>
      <span
        className="Tab__field"
      >
        000.000.000.000:9001
      </span>
    </div>
    <div
      className="row Tab__row"
    >
      <div
        className="ui sub header"
      >
        Hostnames
      </div>
      <span
        className="Tab__field"
      >
        mock-relay.net
      </span>
    </div>
    <div
      className="row Tab__row"
    >
      <div
        className="ui sub header"
      >
        Country
      </div>
      <span
        className="Tab__field"
      >
        Mock
      </span>
    </div>
    <div
      className="row Tab__row"
    >
      <div
        className="ui sub header"
      >
        AS Name
      </div>
      <span
        className="Tab__field"
      >
        Mock Networks
      </span>
    </div>
    <div
      className="row Tab__row"
    >
      <div
        className="ui sub header"
      >
        Consensus weight
      </div>
      <span
        className="Tab__field"
      >
        10000
      </span>
    </div>
    <div
      className="row Tab__row"
    >
      <div
        className="ui sub header"
      >
        Platform
      </div>
      <span
        className="Tab__field"
      >
        Tor 0.3.2.10 on Mock
      </span>
    </div>
    <div
      className="row Tab__row"
    >
      <div
        className="ui sub header"
      >
        Contact
      </div>
      <span
        className="Tab__field"
      >
        Contact
      </span>
    </div>
  </div>
</div>
`;
